dallinger/deployment.py

```diff
@@ -51,10 +50,10 @@ def setup_experiment(log, debug=True, verbose=False, app=None, exp_config=None):
     """Check the app and, if compatible with Dallinger, freeze its state."""
     # Verify that the Postgres server is running.
     try:
-        psycopg2.connect(database="x", user="postgres", password="nada")
-    except psycopg2.OperationalError as e:
-        if "could not connect to server" in str(e):
-            raise RuntimeError("The Postgres server isn't running.")
+        db.check_connection()
+    except Exception:
```

Comment:
It's not the end of the world to catch bare Exception here, as we do reraise it, but I don't think I understand the motive for  loosening it here.

I think I would personally have done:

    except psycopg2.OperationalError as e:
        six.raise_from(RuntimeError("The postgres server isn't running.", e)

but it's fine as it is.

Reply:
Hmm... my goal was to be a little less down in the weeds here by including less specific knowledge about what probably went wrong (the deployment function no longer cares that it's a PG database, even). The only reason to catch the exception is to add the log message to provide a little friendly context to the traceback spew that will follow, and we want to do that regardless of the exception type (I think?)